build.gradle

```diff
@@ -377,19 +453,96 @@ task overkill {
     dependsOn test, test_extra_checks, test_integration, test_integration_extra_checks, dieharder_all
 }
 
+task fakePublish {
+     dependsOn emptyJar
+     outputs.dir("${buildDir}/fake_maven")
+
+    def x64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_x64_Artifacts}/lib"
+    def aarch64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_aarch64_Artifacts}/lib"
+    def newPrefix = "AmazonCorrettoCryptoProvider-${version}"
+    if (isFips) {
+        x64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_x64_Artifacts_FIPS}/lib"
+        aarch64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_aarch64_Artifacts_FIPS}/lib"
+        newPrefix = "AmazonCorrettoCryptoProvider-FIPS-${version}"
+    }
+    def x86Jar = "${x64Base}/AmazonCorrettoCryptoProvider.jar"
+    def aarch64Jar = "${aarch64Base}/AmazonCorrettoCryptoProvider.jar"
+    def sourceJar = "${x64Base}/AmazonCorrettoCryptoProvider-sources.jar"
+    def javadocJar = "${x64Base}/AmazonCorrettoCryptoProvider-javadoc.jar"
+
+    doLast {
+        println "Running on " + osdetector.classifier
+        println aarch64Jar
+        println file(aarch64Jar).file
+        println x86Jar
+        println file(x86Jar).file
+        println sourceJar
+        println file(sourceJar).file
+        println javadocJar
+        println file(javadocJar).file
+
+        def destinationDir = "${buildDir}/fake_maven"
+        copy {
+            from aarch64Jar
+            rename 'AmazonCorrettoCryptoProvider.jar', "${newPrefix}-linux-aarch_64.jar"
+            into "${destinationDir}"
+        }
+        copy {
+            from x64Base
+            rename { String filename ->
+                if (filename.equals("AmazonCorrettoCryptoProvider.jar")) {
+                    return "${newPrefix}-linux-x86_64.jar"
+                } else {
+                    return filename.replace("AmazonCorrettoCryptoProvider-", "${newPrefix}-")
```

Comment:
when would we hit this `else` case?

why is this treated differently from aarch64?

Reply:
We pull all generic (non-platform specific) JARs from the x64 build. Why? They have to come from somewhere, so why not? The else branch is hit for the generic JARs (such as source and javadoc).

aarch64 only pulls the actual ACCP jar.

Reply:
sure, makes sense.